### PR TITLE
NIFI-11758: Added local file upload option in PutAzure*Storage proces…

### PR DESCRIPTION
**Pull request by turcsanyip**
…sors

# Summary

[NIFI-11758](https://issues.apache.org/jira/browse/NIFI-11758)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the contribution @turcsanyip.

Although the Jira issue provides some background on potential use cases, this raises some larger questions about integration patterns for processing local files. The Local File use case presumes some other process is responsible for landing files in a particular location.

One important note, the Local File strategy has security implications, and should require applying the `READ_FILESYSTEM` permission to the associated Processors. That also impacts current behavior.

File: nifi-nar-bundles/nifi-extension-utils/nifi-data-upload/src/main/java/org/apache/nifi/processors/dataupload/DataUploadSource.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.dataupload;
+
+import org.apache.nifi.components.DescribedValue;
+
+public enum DataUploadSource implements DescribedValue {
+
+    FLOWFILE_CONTENT("FlowFile's Content", "The content of the incoming FlowFile will be uploaded."),
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-extension-utils/nifi-data-upload/src/main/java/org/apache/nifi/processors/dataupload/DataUploadSource.java:23]: Recommend avoiding the possessive and just naming this `FlowFile Content` if this goes forward.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@turcsanyip On further consider, the concern regarding filesystem permissions seems to warrant a different approach. Assigning new permissions to these Processors would impact existing deployments, potentially restricting access more than necessary for standard use cases.

As an alternative approach, a new Controller Service interface could be introduced that provides File System access. The general contract could return an `InputStream` from a `java.net.URI` object. The implementation could easily return a `FileInputStream` for a URI-derived Path, and the buffered upload approach could work. This could also support a similar approach for other Processors. The new interface could be named something along the lines of `FileResourceProvider` or perhaps `StreamResourceProvider`, depending on how generic this should be scoped.

The advantage of the new Controller Service interface is that the implementation can be annotated with the Read Filesystem Permission. This avoids impacting existing flows, and makes it clear that this approach requires additional permissions.

If this sounds like a good approach, it may be helpful to close the current PR and open a new one along the lines of the proposed implementation.

**Comment by turcsanyip** [comment]
Thanks @exceptionfactory for sharing your ideas! Closing this PR in favour of a new one based on the suggested concept.

**Pull request closed without merging**